bolt-simple-taproot.md

```diff
@@ -0,0 +1,943 @@
+local commitment and the remote commitment. If only one nonce was used, when a
+party went to sign and broadcast their local commitment, they would re-use a
+nonce, as the nonce was already used to sign the remote party's commitment.
+
+#### `accept_channel` Extensions
+
+1. `tlv_stream`: `accept_channel_tlvs`
+2. types:
+    1. type: 2 (`local_musig2_pubnonce`)
+    2: data:
+        * [`66*byte`:`nonces`]
+    1. type: 4 (`remote_musig2_pubnonce`)
+    2: data:
+        * [`66*byte`:`nonces`]
+
+##### Requirements
+
+The sender:
+
+ - MUST set `local_musig2_pubnonce` and `remote_musig2_pubnonce` to the
+   `musig2` public nonce as specified by the `NonceGen` algorithm of
+   `bip-musig2`
+
+ - Aggregate the received local+remote public nonces (into two combined nonces)
+   using the `NonceAgg` algorithm of `bip-musig2`.
+
+The receiver:
+
+ - MUST reject the channel if `local_musig2_pubnonce` or
+   `remote_musig2_pubnonce` are not present.
+
+#### `funding_created` Extensions
+
+The sender:
+
+  - MUST derive the aggregated `musig2` public key using the exchanged `funding_pubkey`s 
+    - Before aggregating the two funding public keys, the keys MUST be sorted
+      using the `KeySort` routine of `bip-musig2`.
+
+  - MUST use the exchanged `local_musig2_pubnonce` and `remote_musig2_pubnonce`
+    received from the remote party with the `NonceAgg` algorithm to derive the
+    combined nonce
+
+  - the `signature` field MUST be a `musig2` partial signature using the
+    aggregated key defined above, and the two public nonces exchanged in prior
+    messages.
+
+  - MUST generate the attached `signature` using the combined
+    `remote_musig2_pubnonce` it generated in the prior step.
+
+
+The recipient:
+  - MUST validate the signature as a _partial_ signature based on the
+    `PartialSigVerifyInternal` (using the combined nonce for the remote party's
+    commitment) algorithm of `bip-musig2`:
+
+    - if the partial signature is invalid, MUST fail the channel
+
+#### `funding_signed` Extensions
+
+The sender MUST set:
+
+  - `signature` to the valid signature, using its `funding_pubkey` for the
+    initial commitment transaction, as defined in this documewnt.
+    - this signature MUST be a partial signature as defined by `bip-musig2`.
+
+The recipient:
+
+  - MUST validate the signature as a _partial_ signature based on the
+    `PartialSigVerifyInternal` algorithm of `bip-musig2`:
+
+    - if the partial signature is invalid, MUST fail the channel
+
+#### `funding_locked` Extensions
+
+We add a new TLV field to the `funding_locked` message:
+
+1. type: 36 (`funding_locked`)
+2. data:
+    * ...
+    * [`funding_locked_tlvs`:`tlvs`]
+
+1. `tlv_stream`: `funding_locked_tlvs`
+2. types:
+    1. type: 0 (`local_musig2_pubnonce`)
+    2. data:
+        * [`musig2_pubnonce`:`nonces`]
+    1. type: 2 (`remote_musig2_pubnonce`)
+    2. data:
+        * [`musig2_pubnonce`:`nonces`]
```

Comment:
These nonces only need to sent the first time `funding_locked` is sent iff a `channel_reestablish` has not already been exchanged. With #910 , the `funding_locked` message would continually replace the nonces, possibly messing up the channel state since the sender of `funding_locked` may not know whether a received `commit_sig` is using the right nonces due to the lack of synchronization. So I think these nonces should actually be exchanged in `funding_created`/`funding_signed`

Reply:
Good point, I was thinking of the case where a party sends `funding_locked` more than once to update/refresh their scid aliases... 

The goal here was to mirror the existing `funding_locked` behavior and disallow creating a new commitment until this is sent. However this message still has the next revocation key, so you can't make a state w/o that anyway. 

But if we removed the nonces here, then both sides would need to remember the nonce state between restarts. So we might _just_ want to specify that the nonces only need to be in place the very first time chan reest was sent.